src/generateSdk/sdkWriter/jsSdkWriter.ts

```diff
@@ -0,0 +1,165 @@
+import path from "path";
+import { default as fsExtra } from "fs-extra";
+import fsPromises from "fs/promises";
+import fs from "fs";
+import { createLocalTempFolder, deleteFolder } from "../../utils/file.js";
+import { ClassUrlMap, replaceUrlsInSdk, writeSdkToDisk } from "../../utils/sdk.js";
+import { getNodeModulePackageJson } from "../templates/packageJson.js";
+import { compileSdk } from "../utils/compileSdk.js";
+import { Language } from "../../yamlProjectConfiguration/models.js";
+import { SdkGeneratorResponse } from "../../models/sdkGeneratorResponse.js";
+import { getLinkPathsForProject } from "../../utils/linkDatabase.js";
+import { debugLogger } from "../../utils/logging.js";
+import { YamlFrontend } from "../../yamlProjectConfiguration/v1.js";
+import { Options, Result, compareSync } from "dir-compare";
+
+const POLLING_INTERVAL = 2000;
+
+async function writeSdk(projectName: string, frontend: YamlFrontend|undefined, projectRegion: string, stage: string, sdkResponse: SdkGeneratorResponse, classUrls: ClassUrlMap[], publish: boolean, language: Language) {
+    await replaceUrlsInSdk(
+        sdkResponse,
+        classUrls,
+    );
+
+    const localPath = await createLocalTempFolder(
+        `${projectName}-${projectRegion}`,
+    );
+    const sdkPath = path.join(localPath, "sdk")
+    await deleteFolder(sdkPath);
+    await writeSdkToDisk(sdkResponse, sdkPath);
+    const packageJson: string = getNodeModulePackageJson(
+        /* packageName= */ `@genezio-sdk/${projectName}_${projectRegion}`,
+        /* packageVersion= */ `1.0.0-${stage}`,
+    );
+    await compileSdk(sdkPath, packageJson, language, publish);
+
+    if (stage === "local") {
```

Comment:
I will remove this from here. It will be an "sdk monitor" component which `local.ts` will invoke on all languages. All of them except js and ts will have a no-op.